### DISCUSSION TITLE
MQTT and Home Assistant: How to register the UZG into HomeAssistant

### DISCUSSION THREAD

**Discussion by Jeppedy** [xyzroe/XZG#89] [Q&A]
So, I turned on and connected the UZG-01 via MQ to my MQTT Broker.  I can see the messages posting to the queue in the homeassistant topic.  But Home Assistant isn't discovering the UZG automatically, and I have no idea how to add it manually.
Any suggestions?  Ideally, I'd love to see documentation that speaks to this.

**Comment by xyzroe** [comment]
All you need to do is enable MQTT and the Home Assistant (MQTT Discovery) option on the MQTT page of your UZG gateway. There isn’t much to document regarding how to enable it, but a more detailed description is available here: https://xzg.xyzroe.cc/mqtt/. 
I would appreciate any additional information you can add to the Wiki.

**Comment by Jeppedy** [comment]
Thanks!  Once I enable it, should I expect to see the UZG appear as a Z2M
Zigbee device in the Z2M UI?

**Reply by xyzroe** [reply]
No, of course no. 
You need more to learn about how zigbee2mqtt, MQTT, Home Assistant works.

**Comment by Jeppedy** [comment]
Actually, that seems to be almost what happens.  Except, I expected it to
tell me it found a device, and instead, it just appears in my device list.

See, here's the thing.  I'm actually very technical.  And I've been a deep
HA user for over a year.  I have Proxmox in my lab, a Pi for HA, etc.
But I think there's a layer of information missing around Z2M.  Not deep
techie info about Zigbee protocols, just info like what Z2M is looking for
in the queues, why I can't leave Z2M running when I update the coordinator
(if it just reads the Zigbee msgs and there aren't any during update, why
does it care), etc.
There are no clear help guides for debugging (if it's seeing updates in
MQTT, why does HA show my device as updated two minutes ago AND offline at
the same time?)
BTW, that was the whole source of my confusion.  I was desperately trying
to debug, and it turns out if I left it overnight, my network self-healed
(Why were my Last Seen values updating for offline devices??)

**Reply by xyzroe** [reply]
I am sure that there is more than enough information on how to use it. The questions you ask are not related to use, but are rather of practical interest to you personally. If you had followed the guide for installing and configuring z2m and updating the coordinator firmware, then you would not have encountered half the problems. 
And the fact that you are asking questions about why the device in MQTT and HA has different statuses online and the last update time was 2 minutes ago, while online says that in the theoretical part of using HA and MQTT you also have some unstudied sections. 
Please follow the [getting started guide](https://www.zigbee2mqtt.io/guide/getting-started/) step by step. Than all other pages and you will find answers on all your questions. But you need to read it carefully. Not all information is written explicitly.

**Comment by Jeppedy** [comment]
You are right.  They are of personal interest... I have a running instance
with 40+ devices.  I'm not starting fresh.
I was upgrading from a Sonoff zigbee dongle to the networked UZG.
Worked great, until the upgrade when all the devices reported offline, yet
reporting in actively.  And my timeout setting is far more than 2 minutes
for battery devices.
I thank you for the doc reference.  I've read it and the other website
pages many times already, but am glad to read it again.  Perhaps I've
missed the background info.
I'm not trying to be critical.  I'm trying to give you a new perspective
from a technical person who is new to Z2M details.  I'll stop bothering you
now.  Perhaps over time, documentation will naturally grow.
